app/Filter/Forms/Types/AuthoritySearchRequestType.php

```diff
@@ -46,8 +46,8 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             $searchRequest = $event->getData();
             $form = $event->getForm();
 
-            $getChoiceOptions = function ($attribute) use ($searchRequest) {
-                $choices = $this->authorityRepository->listValues($attribute, $searchRequest);
+            $getChoiceOptions = function ($attribute, $translation_domain = null) use ($searchRequest) {
+                $choices = $this->authorityRepository->listValues($attribute, $searchRequest, null, $translation_domain);
```

Comment:
This should probably be in camelCase (as other vars are)
```suggestion
            $getChoiceOptions = function ($attribute, $translationDomain = null) use ($searchRequest) {
                $choices = $this->authorityRepository->listValues($attribute, $searchRequest, null, $translationDomain);
```

But I'm actually confused by the term "translation domain" entirely :D Is it not the same as "locale" or "lang"?

```suggestion
            $getChoiceOptions = function ($attribute, $locale = null) use ($searchRequest) {
                $choices = $this->authorityRepository->listValues($attribute, $searchRequest, null, $locale);
```

Reply:
I decided to get ispired by the naming/convention in [symfony forms](https://symfony.com/doc/current/reference/forms/types/form.html#translation-domain)
Initially I wanted to use it directly - since there is `choice_translation_domain` option available (but it's not possible to use in in our case - since we mix up the choice values with counts)
We already use this option e.g. in https://github.com/SlovakNationalGallery/webumenia.sk/blob/feature/WEBUMENIA-1584-add-gender-filter-other-variant/app/Filter/Forms/Types/AuthoritySearchRequestType.php#L92

But now I see - there it's option. Here it's variable name. Do you think it would be better to use camelCase?

Reply:
On a second look, I'm wondering whether the translation domain should be mentioned in repositories at all. Something like `authorityRepository->listValues` should probably not be concerned with the presentation at all, IMHO. Same for `createBucketCollection()`. I think we're mixing in presentation concerns too soon.

Another thing is that `TranslatableRepository->listValues` now has both `$locale` and `$translation_domain`. That makes things a bot confusing, doesn't it? Now I see that this was probably originally designed to store localized strings in the index as proposed in #596.

My proposal would be to limit the translation domain logic only to this function `$getChoiceOptions`. We'd probably have to refactor the `listValues` to return something like 
```php
[
  [
    'value' => 'ABC',
    'count' => 33,
  ],
  [
    'value' => 'XYZ',
    'count' => 14,
  ],
]
```
and to turn that into actual [choices](https://symfony.com/doc/current/reference/forms/types/choice.html#choices) here, in `$getChoiceOptions`. 

That would reduce the presentation logic "creep" from Repositories and it would give us more flexibility to translate closer to the front-end.

Reply:
Also, if you'll want to revert to #596, that would be fine by me too. It would be the cheaper/faster option.